src/actions/hubspot/hubspot.ts

```diff
@@ -0,0 +1,257 @@
+import * as hubspot from "@hubspot/api-client"
+import * as semver from "semver"
+import * as util from "util"
+import * as winston from "winston"
+import * as Hub from "../../hub"
+import { RequiredField } from "../../hub"
+import { HubspotActionError } from "./hubspot_error"
+
+export enum HubspotTags {
+  ContactId = "hubspot_contact_id",
+  CompanyId = "hubspot_company_id",
+}
+
+export enum HubspotCalls {
+  Contact = "contact",
+  Company = "company",
+}
+
+interface DefaultHubspotConstructorProps {
+  name: string
+  label: string
+  description: string
+  call: HubspotCalls
+  tag: HubspotTags
+}
+
+async function delay(ms: number) {
+  // tslint continually complains about this function, not sure why
+  // tslint:disable-next-line
+  return new Promise((resolve) => setTimeout(resolve, ms))
```

Comment:
it's probably mad about not having a reject or .catch function

Reply:
Yeah, I tried that but still receiving: `Explicit type parameter needs to be provided to the constructor`